### PR TITLE
Replaced obsolete foreach-loops with for-loops

### PR DESCRIPTION
**Pull request by alexeylysenko**
As requested here https://github.com/owncloud/client/pull/8700#issuecomment-856833445 foreach-loops have been replaced with for-loops.
However, the issue with unnecessary temporary containers is still covered here https://github.com/owncloud/client/pull/8700

### CODE REVIEW

File: src/common/ownsql.cpp [hunk 1]
```diff
@@ -190,7 +190,9 @@ QString SqlDatabase::error() const
 void SqlDatabase::close()
 {
     if (_db) {
-        for (auto *q : qAsConst(_queries)) {
+        // we need a copy because query is removed from a set in finish()
+        const auto queries = _queries;
```
Comment (TheOneRing) [inline src/common/ownsql.cpp:194]: how did you spot this one?

File: src/gui/activitywidget.cpp [hunk 2]
```diff
@@ -588,8 +588,8 @@ void ActivitySettings::slotRefresh(AccountState *ptr)
 
 void ActivitySettings::slotRegularNotificationCheck()
 {
-    AccountManager *am = AccountManager::instance();
-    foreach (AccountStatePtr a, am->accounts()) {
+    const auto accounts = AccountManager::instance()->accounts();
```
Comment (TheOneRing) [inline src/gui/activitywidget.cpp:591]: Its not obvious that in that case accounts  is a const reference so it looks like an explicit copy for no reason.
Any reason for that, besides an unrequired limit of 80 chars :)

File: src/common/ownsql.cpp (hunk 1 above)
Comment (alexeylysenko) [reply src/common/ownsql.cpp:194]: The tests were failed with the first commit. And after brief investigation I found this code
```
void SqlQuery::finish()
{
    ...
    if (_sqldb) {
        _sqldb->_queries.remove(this);
    }
}
```

File: src/gui/activitywidget.cpp (hunk 2 above)
Comment (alexeylysenko) [reply src/gui/activitywidget.cpp]: You're completely right, I pushed the fix.

**Comment by TheOneRing** [comment]
Btw are you interested in joining our rocket chat talk.owncloud.com/

**Comment by alexeylysenko** [comment]
> Btw are you interested in joining our rocket chat talk.owncloud.com/

Yes, I am. Btw, are you able to restart CI job on this pull request? It has failed with 
```/bin/sh: 35: cd: can't cd to docs/```

**Comment by TheOneRing** [comment]
I can but after the recent merges I'd recommend to rebase and force push :)

**Comment by alexeylysenko** [comment]
I did merge from master

**Pull request closed without merging**